pkg/kube/podtemplatespec/podspec_template.go

```diff
@@ -107,6 +107,22 @@ func WithServiceAccount(serviceAccountName string) Modification {
 	}
 }
 
+// WithVolumes appends the given volumes to the existing volume, it ensures not
+// to override existing volumes
```

Comment:
Does it make sense to ensure we don't override volumes? I would think it might be desirable to be able to replace volumes with the new ones.

Reply:
I don't have a strong opinion on this -- I followed what we did for `WithVolume` -- https://github.com/mongodb/mongodb-kubernetes-operator/blob/44c20125f4eb7cff9525e6bfb8259a255c2319c2/pkg/kube/podtemplatespec/podspec_template.go#L127, looks like we skip setting it if it exists. Also users, might configure their own volume, so I am mildly in favour of not overriding.

Reply:
SGTM